### PR TITLE
Added `ui.search.mode` to `.quilt/catalog/config.yaml`

### PR DESCRIPTION
**Pull request by fiskus**
* Added `ui.search.mode` to .quilt/catalog/config.yaml
* Refactored BucketPreferences.ts: added AsyncResult to return value. Used only in new code yet

- [x] Documentation
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

File: catalog/app/containers/Bucket/Search.js [hunk 1]
```diff
@@ -339,15 +342,33 @@ export default function BucketSearch({
 }) {
   const isInStack = BucketConfig.useIsInStack()
   const { q: query = '', p, mode, ...params } = parseSearch(l.search)
+  const { preferences: prefs, result } = BucketPreferences.use()
   const page = p && parseInt(p, 10)
   const retry = (params.retry && parseInt(params.retry, 10)) || undefined
+  console.log(result, prefs)
```
Comment (nl0) [inline catalog/app/containers/Bucket/Search.js:348]: (:

**Review by nl0 — APPROVED** [review]
this only works for in-bucket search, not for global search -- is this intentional?

File: catalog/app/containers/Bucket/Search.js (hunk 1 above)
Comment (fiskus) [reply catalog/app/containers/Bucket/Search.js]: Hmm, with new setup husky+lint works for me. Don't know how it was sliped here

**Comment by fiskus** [comment]
Yes, it's intentional on my side that search mode only works for bucket. But I didn't get feedback from Rob or Aneesh yet

File: catalog/app/containers/Bucket/Search.js [hunk 2]
```diff
@@ -339,6 +342,7 @@ export default function BucketSearch({
 }) {
   const isInStack = BucketConfig.useIsInStack()
   const { q: query = '', p, mode, ...params } = parseSearch(l.search)
+  const { preferences: prefs, result } = BucketPreferences.use()
```
[the remark this answers is not in the document]
Comment (fiskus) [reply catalog/app/containers/Bucket/Search.js:345]: It's not used. Now, I'm 100% sure that husky + eslint works only partially :man_shrugging:

**Review by nl0 — APPROVED** [review]

**Comment by fiskus** [comment]
Re-implemented here https://github.com/quiltdata/quilt/pull/3270

**Pull request closed without merging**